### ISSUE TITLE
Implement option of reducing wait

### ISSUE THREAD

**Issue by 4rm4** [traa-pixelcanvas/TraanvasBot#12]
Implement option to change the waiting time of a pixel, (I know that PC could update the code to detect it and block users, so that need be a optional feature with warnings)

#in bot.py
        if data['waitSeconds'] is not None:
            wait = data['waitSeconds'] + (random.randint(0, 9) / 10.0)
            wait = wait - (self.reduce_time if self.reduce_time < wait else 0)
            print(I18n.get('Waiting %s seconds') % str(wait))

#in main.py 

    parser.add_argument('-rt', '--reduce_time', required=False, type=float, default=3, dest='reduce_time', help='Decrease the cooldown counting by X seconds Ex: 3.3')
``

**Comment by olegispe** [comment]
This can be easily done, but the actual benefit of it confuses me. Pixel wait times are controlled by the site, if we request to early we're just going to get told to wait.

Unless I've misunderstood, this provides no benefit, and actually just causes more errors.

**Comment by aud-tree** [comment]
The problem is that the bot adds a random time between 1.5-3s (I think) to add to the server-generated wait time. Reducing or eliminating that generally works out because network latency is enough to avoid errors, and the worst case is having to retry once after (preferably) a 0.5 sec wait. In a defense situation, this could make the difference between winning and losing.

**Comment by olegispe** [comment]
I guess, I always assumed that was a client side issue with the site, never looked into it myself. In that case, that could work. We'd need to be sure we don't request too early however. A somewhat "safe" reduction in the cooldown?

**Comment by aud-tree** [comment]
The idea is to make this optional, since it's not generally needed and getting 400s from the server might raise a flag on arkeros' end (maybe, who knows). It's up for debate how much the randomness even matters, since I doubt that's what he's using to figure out if someone's a bot. It should technically be impossible to request too early anyway, since we get the wait time X ms after the timer starts on the server, wait the whole time, and then the server gets our next request about X ms after the timer ends on our end, for a buffer of 2X ms, if that makes sense.

**Comment by Mismatchia** [comment]
I'm thinking of writing a function to run at startup that tries to guess the difference between the wait time given by the server and the actual wait time. It would lower the value until it gets an error, then use that modifier on future waits.

For efficiency, we need to find a base value to start guessing from that works for everyone, instead of starting the guess too high.

It should also attempt to monitor the 'too early' errors that it encounters as it goes, and raise the wait time if it starts getting too many.

Finally, it should only react to an error that corresponds with trying to paint too early, not any other sort of connection issues. 

I may also take a look at the timer function to ensure accuracy.

**Comment by Mismatchia** [comment]
This entire process should require no user interaction, and might be something that can use a `--verbose` flag to keep background details from spamming the average user.

**Comment by olegispe** [comment]
Verbosity in that regard can be easily done:

```python
if args.verbose:
    logger.setLevel(logging.DEBUG)
else:
    logger.setLevel(logging.INFO)
```

and then basically change the more detailed logs to `logger.debug` and general logs to `logger.info`.

**Comment by Mismatchia** [comment]
Good thinking! Would you mind opening a `--verbose` enhancement issue?

**Comment by aud-tree** [comment]
Implemented in #71
